src/eureka/S3_data_reduction/plots_s3.py

```diff
@@ -1289,33 +1291,39 @@ def tilt_events(meta, data, m, position, median_tilt_frame):
         images.append(imageio.v2.imread(meta.outputdir + fname))
 
     # Figure fig3507b
-    # Create .gif per segment
-    for i in tqdm(range(1), desc='  Creating segment tilt event GIF'):
-        imageio.mimsave(meta.outputdir + f'figs{os.sep}' +
-                        f'fig3507b_tilt_event_segment_{file_number}.gif', 
-                        images, fps=20)
+    # Create .gif per batch
+    log.writelog('  Creating batch tilt event GIF',
+                 mute=(not meta.verbose))
+    imageio.mimsave(meta.outputdir + f'figs{os.sep}' +
+                    f'fig3507b_tilt_event_batch_{file_number}.gif', 
+                    images, fps=20)
 
     # Figure fig3507c
     # Create .gif of all tilt event segments combined
-    if meta.testing_S3 is False and (m + 1 is meta.num_data_files):
-        for i in tqdm(range(1), desc='  Creating all segment tilt event GIF'):
-
-            # Create list of all .png tilt images in tilt_event folder
-            all_images = []
-            in_filenames = os.listdir(meta.outputdir + 
-                                      f'figs{os.sep}tilt_events{os.sep}')
-            in_filenames.sort()
-
-            # Create list of all figure names to pull from later to create .gif
-            for i in range(0, len(in_filenames)):
-                all_images.append(imageio.v2.imread(meta.outputdir + 
-                                                    f'figs{os.sep}' + 
-                                                    f'tilt_events{os.sep}' + 
-                                                    in_filenames[i]))
-
-            # Create .gif of all tilt event segments
-            imageio.mimsave(meta.outputdir + f'figs{os.sep}' +
-                            'fig3507c_tilt_events_all_segments.gif', 
-                            all_images, fps=60)
-
-    return median_frame
+    if not meta.testing_S3 and (m + 1 == meta.nbatch):
+        log.writelog('  Creating all segment tilt event GIF',
+                     mute=(not meta.verbose))
+
+        # Create list of all .png tilt images in tilt_event folder
+        all_images = []
+        for file in os.listdir(meta.outputdir + 
+                               f'figs{os.sep}tilt_events{os.sep}'):
+            if file.endswith(".png"):
+                in_filenames = os.path.join(meta.outputdir + 
+                                            f'figs{os.sep}' + 
+                                            f'tilt_events{os.sep}', file)
+        in_filenames.sort()
```

Comment:
`in_filenames` gets over-written every time it finds a new png which is definitely not your intended behaviour. You're going to need to append the result of `os.path.join` to some list